recipes/boost/all/conanfile.py

```diff
@@ -1246,45 +1266,69 @@ def add_libprefix(n):
+
+                self.cpp_info.components[module].libs = module_libraries
+
+                self.cpp_info.components[module].requires = self._dependencies["dependencies"][module] + ["_libboost"]
+                self.cpp_info.components[module].names["cmake_find_package"] = module
+                self.cpp_info.components[module].names["cmake_find_package_multi"] = module
+                self.cpp_info.components[module].names["pkg_config"] = "boost_{}".format(module)
+
+                for requirement in self._dependencies.get("requirements", {}).get(module, []):
+                    if requirement == "backtrace":
+                        # FIXME: backtrace not (yet) available in cci
+                        continue
+                    if self.options.get_safe(requirement, None) == False:
+                        continue
+                    conan_requirement = self._option_to_conan_requirement(requirement)
+                    if conan_requirement not in self.requires:
+                        continue
+                    if module == "locale" and requirement in ("icu", "iconv"):
+                        if requirement != self.options.i18n_backend:
                             continue
-                        conan_requirement = self._option_to_conan_requirement(requirement)
-                        if conan_requirement not in self.requires:
-                            continue
-                        if module == "locale" and requirement in ("icu", "iconv"):
-                            if requirement != self.options.i18n_backend:
-                                continue
-                        self.cpp_info.components[module].requires.append("{0}::{0}".format(conan_requirement))
+                    self.cpp_info.components[module].requires.append("{0}::{0}".format(conan_requirement))
+
+            for incomplete_component in incomplete_components:
+                self.output.warn("Boost component '{0}' is missing libraries. Try building boost with '-o boost:without_{0}'.".format(incomplete_component))
 
-            if used_libraries != detected_libraries:
-                non_used = detected_libraries.difference(used_libraries)
-                assert len(non_used) == 0, "These libraries were not used in conan components: {}".format(non_used)
+            non_used = all_detected_libraries.difference(all_expected_libraries)
+            if non_used:
+                raise RuntimeError("These libraries were built, but were not used in any boost module: {}".format(non_used))
 
-                non_existing = used_libraries.difference(detected_libraries)
-                assert len(non_existing) == 0, "These libraries were used, but not built: {}".format(non_existing)
+            non_built = all_expected_libraries.difference(all_detected_libraries)
+            if non_built:
+                raise RuntimeError("These libraries were expected to be built, but were not built: {}".format(non_built))
```

Comment:
```suggestion
                raise ConanException("These libraries were expected to be built, but were not built: {}".format(non_built))
```
Use `ConanException` instead. It's filtered by Conan and provides a better output.

Reply:
Good point, thanks for the review.

Please review again :stuck_out_tongue: